File: elixir/06-retirement_calculator/retirement_calculator.ex
Code.load_file("../util/input.ex")

defmodule RetirementCalculator do
  import Input

  def run do
    age = int_p("What is your current age? ")
    retirement_age = int_p("What age would you like to retire? ")

    print_result(retirement_age - age)
  end

  defp print_result(time_ultil_retire) when time_ultil_retire > 0 do
    current_year = Date.utc_today.year
    IO.puts "You have #{time_ultil_retire} years left until you can retire.\n" <>
            "It's #{current_year}, so you can retire in #{current_year + time_ultil_retire}"
  end

  defp print_result(_time_ultil_retire), do: IO.puts "You can already retire."
end

RetirementCalculator.run()

File: python/06-retirement_calculator/retirement_calculator.py
import datetime
import textwrap

def print_result(time_until_retire):
    current_year = datetime.date.today().year

    if time_until_retire > 0:
        print(textwrap.dedent(
            f"""\
            You have {time_until_retire} years left until you can retire.
            It's {current_year}, so you can retire in {current_year + time_until_retire}.
            """))
    else:
        print("You can already retire.")

age = int(input("What is your current age? "))
retirement_age = int(input("What age would you like to retire? "))

print_result(retirement_age - age)
